File: ch5/clip3/boj2096/main.py
import sys

n = int(sys.stdin.readline())
MAX_VALUE = 1000000

dmx = [0] * 5
dmn = [0] * 5
tmx = [0] * 5
tmn = [0] * 5

dmx[0] = dmx[4] = 0
dmn[0] = dmn[4] = MAX_VALUE

for _ in range(n):
    inputs = list(map(int, sys.stdin.readline().split()))
    for j in range(1, 4):
        input_value = inputs[j - 1]
        tmx[j] = max(dmx[j - 1], max(dmx[j], dmx[j + 1])) + input_value
        tmn[j] = min(dmn[j - 1], min(dmn[j], dmn[j + 1])) + input_value

    for j in range(1, 4):
        dmx[j] = tmx[j]
        dmn[j] = tmn[j]

max_score = max(dmx[1:4])
min_score = min(dmn[1:4])

print(max_score, min_score)

File: ch5/clip3/boj2096/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        final int MAX_VALUE = 1000000;
        int input = 0;
        int dmx[] = new int[5];
        int dmn[] = new int[5];
        int tmx[] = new int[5];
        int tmn[] = new int[5];
        dmx[0] = dmx[4] = 0;
        dmn[0] = dmn[4] = MAX_VALUE;
        for(int i = 1; i <= n; i++) {
            for(int j = 1; j <= 3; j++) {
                input = sc.nextInt();
                tmx[j] = Math.max(dmx[j - 1], Math.max(dmx[j], dmx[j + 1])) + input;
                tmn[j] = Math.min(dmn[j - 1], Math.min(dmn[j], dmn[j + 1])) + input;
            }
            for(int j = 1; j <= 3; j++) {
                dmx[j] = tmx[j];
                dmn[j] = tmn[j];
            }
        }
        int max = 0;
        int min = MAX_VALUE;
        for(int i = 1; i <= 3; i++) {
            max = Math.max(max, dmx[i]);
            min = Math.min(min, dmn[i]);
        }
        System.out.println(max + " " + min);
    }
}


